resources/fetchers/kube_factory.go

```diff
@@ -40,16 +41,16 @@ func (f *KubeFactory) Create(log *logp.Logger, c *config.C, ch chan fetching.Res
 	if err != nil {
 		return nil, err
 	}
-
-	return f.CreateFrom(log, cfg, ch)
+	return f.CreateFrom(log, cfg, ch, kubernetes.GetKubernetesClient)
 }
 
-func (f *KubeFactory) CreateFrom(log *logp.Logger, cfg KubeApiFetcherConfig, ch chan fetching.ResourceInfo) (fetching.Fetcher, error) {
+func (f *KubeFactory) CreateFrom(log *logp.Logger, cfg KubeApiFetcherConfig, ch chan fetching.ResourceInfo, provider func(kubeconfig string, opt kubernetes.KubeClientOptions) (k8s.Interface, error)) (fetching.Fetcher, error) {
```

Comment:
It would be nice if you could create a type for this func, will be more readable.
Something like: `type K8sProviderFunc func(string, kubernetes.KubeClientOptions) (k8s.Interface, error)`